src/mongocrypt-crypto.c

```diff
@@ -299,7 +299,6 @@ bool _mongocrypt_hmac_sha_256(_mongocrypt_crypto_t *crypto,
                               const _mongocrypt_buffer_t *in,
                               _mongocrypt_buffer_t *out,
                               mongocrypt_status_t *status) {
-    BSON_ASSERT_PARAM(crypto);
```

Comment:
- When would calling `_mongocrypt_hmac_sha_256` with `crypto == nullptr` not be a bug? I imagine there's some use case for this in the server code that's the reason for this ticket, but in libmongocrypt itself, I think that would always be a bug.
- Why doesn't this apply to the other functions in this file that use the hook if provided and fall back to the native implementation otherwise?

Reply:
There's a rather simple workaround that we can do in the server code for this specific case. I think it would be better to do that than to change the behaviors/expectations of multiple functions in `libmongocrypt`. Thanks for bringing up your concerns.